extensions/wikia/UserLogin/js/FacebookLogin.js

```diff
@@ -287,6 +287,30 @@
 					}
 				}
 			});
+
+			this.initSignupFormValidation();
+		},
+
+		initSignupFormValidation: function () {
+			var validator,
+				wikiaForm = this.signupForm.wikiaForm,
+				inputs = wikiaForm.inputs,
+				inputsToValidate = ['username', 'password'];
+
+			if (inputs.email) {
+				inputsToValidate.push('email');
+			}
+
+			validator = new UserSignupAjaxValidation({
+				wikiaForm: wikiaForm,
+				inputsToValidate: inputsToValidate,
+				submitButton: inputs.submit
+			});
+
+			// Add validation on blur event for all inputs to validate
+			inputsToValidate.forEach(function (inputName) {
+				inputs[inputName].on('blur', validator.validateMappedInput.bind(validator));
```

Comment:
Yeah, this joining selectors - although it's easy, it is also unreadable in this particular case.
And maybe sth like that:
```javascript
filtered = $();
inputsToValidate.forEach(function (inputName) {
   filtered.add (inputs[inputName]);
})
filtered.on ('blur', validate);
```

Loop is similar, but only one binding.